### PR TITLE
Update SPRING_APPLICATION_JSON examples to structured format

### PR DESCRIPTION
**Pull request by dturanski**
I verified these samples do not result in errors with 
 -  `kubectl` edit K8s SCDF deployment spec
 -  `cf push` with PCF manifest,
 - `export` and `cf set-env`

Did not verify detailed configuration re kafka, influx, grafana, etc.

### CODE REVIEW

File: content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md [hunk 1]
```diff
@@ -140,29 +144,67 @@ Complete example of a DataFlow manifest that enables metrics collection for both
...
+      SPRING_CLOUD_SKIPPER_CLIENT_SERVER_URI: http://your-skipper-server-uri/api
+      SPRING_APPLICATION_JSON: |-
+        {   
+            "maven" : {
+                "remoteRepositories": {
+                    "repo1" : {
+                        "url": "https://repo.spring.io/libs-snapshot"
+                    }
+                }
+            }
```
Comment (sabbyanandan) [inline content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md]: Don't we need a `,` here and rest of the separate sections below?

File: content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md [hunk 2]
```diff
@@ -140,29 +144,67 @@ Complete example of a DataFlow manifest that enables metrics collection for both
...
+            "spring.cloud.dataflow.task.platform": {
+                "cloudfoundry": {
+                    "accounts": {
+                        "default": {
+                            "connection":{
+                                "url": <api-url>
```
Comment (sabbyanandan) [inline content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md]: Same here. Don't we need `,`s for every line items from here onwards?

File: content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md [hunk 3]
```diff
@@ -140,29 +144,67 @@ Complete example of a DataFlow manifest that enables metrics collection for both
...
+                                "org": <org>
+                                "space": <space>
+                                "domain": <app-domain>
+                                "username": <email>
+                                "password": <password>
+                                "skipSslValidation": true
+                            }
```
Comment (sabbyanandan) [inline content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md]: `,` here also - no?

File: content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md [hunk 4]
```diff
@@ -55,8 +55,20 @@ applications:
...
+        {
+            "spring.cloud.stream.kafka.binder": {
+                   "brokers": "foo0.broker.foo,foo1.broker.foo,foo2.broker.foo",
+                   "jaas.options": {
+                          "username": "test",
+                          "password":"bestest"
+                   },
+                   "jaas.loginModule":"org.apache.kafka.common.security.plain.PlainLoginModule"
+                   }
+                },
```
Comment (sabbyanandan) [inline content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: I don't think we need `,` here.

File: content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md [hunk 5]
```diff
@@ -55,8 +55,20 @@ applications:
...
+            }
```
Comment (sabbyanandan) [inline content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: `,` here, though.

File: content/documentation/pages/1-installation/3-kubernetes/1-helm.md [hunk 6]
```diff
@@ -556,7 +556,17 @@ do so:
 ```yaml
 env:
   - name: SPRING_APPLICATION_JSON
-    value: '{ "maven": { "local-repository": null, "remote-repositories": { "repo1": { "url": "https://repo.spring.io/libs-snapshot"} } } }'
+    value: -|
```
Comment (sabbyanandan) [inline content/documentation/pages/1-installation/3-kubernetes/1-helm.md]: I think it should be: `value: |-` instead.

File: content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md [hunk 7]
```diff
@@ -140,29 +144,67 @@ Complete example of a DataFlow manifest that enables metrics collection for both
...
-    "spring.cloud.dataflow.grafana-info.url": "https://grafana-uri:port"}'
+  - name: data-flow-server
+    host: data-flow-server
+    memory: 2G
+    disk_quota: 2G
+    instances: 1
+    path: ./spring-cloud-dataflow-server-2.2.0.BUILD-SNAPSHOT.jar
```
Comment (sabbyanandan) [inline content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md]: Perhaps we could switch to pull the version from the `variables.json` file as we do in the `wget` commands.

File: content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md [hunk 8]
```diff
@@ -140,29 +144,67 @@ Complete example of a DataFlow manifest that enables metrics collection for both
...
+                "task.management.metrics.export.influx": {
+                    "enabled": true,
+                    "db": "defaultdb",
+                    "autoCreateDb": false,
+                    "uri": "https://influx-uri:port",
+                    "userName": "guest",
+                    "password": "******"
+                },
+                "stream.management.metrics.export.influx": {
+                     "enabled": true,
```
Comment (sabbyanandan) [inline content/documentation/pages/5-feature-guides/2-batch/4-monitoring.md]: extra space here

File: content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md [hunk 9]
```diff
@@ -83,27 +93,55 @@ With that flexibility, every stream application deployed through SCDF will autom
...
-  timeout: 180
-  instances: 1
-  path: spring-cloud-dataflow-server-2.1.0.RELEASE.jar
+  - name: scdf-server
+    host: scdf-server
+    memory: 2G
+    disk_quota: 2G
+    timeout: 180
+    instances: 1
+    path: spring-cloud-dataflow-server-2.1.0.RELEASE.jar
```
Comment (sabbyanandan) [inline content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: Same as above. It'd be nice to pull the latest version from the `variables.json`

File: content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md [hunk 10]
```diff
@@ -83,27 +93,55 @@ With that flexibility, every stream application deployed through SCDF will autom
...
+                        },
+                        "deployment": {
+                            "services": <comma delimited list of service>"
+                        }
+                    }
+                }
+            }, 
+            "stream": { 
+                "kafka.binder": {
+                    brokers": "foo0.broker.foo,foo1.broker.foo,foo2.broker.foo",
```
Comment (sabbyanandan) [inline content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: `"` missing in the beginning

File: content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md [hunk 11]
```diff
@@ -83,27 +93,55 @@ With that flexibility, every stream application deployed through SCDF will autom
...
+                        "options": {
+                            "username": "test",
+                            "password":"bestest"
+                        },
+                        "loginModule":"org.apache.kafka.common.security.plain.PlainLoginModule"
+                    }
+                },
+                "bindings.output.destination":"fooTopic"
+            },
+            "skipper.client.serverUri": "https://<SKIPPER_URI>/api"
```
Comment (sabbyanandan) [inline content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: We could follow the same format that we do in the previous sections for consistency.

> SPRING_CLOUD_SKIPPER_CLIENT_SERVER_URI: http://your-skipper-server-uri/api

File: content/documentation/pages/1-installation/3-kubernetes/1-helm.md (hunk 6 above)
Comment (dturanski) [reply content/documentation/pages/1-installation/3-kubernetes/1-helm.md]: Hmmm, that's what's in my local copy

File: content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md (hunk 11 above)
Comment (dturanski) [reply content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: Do you mean the value or use separate env var for this?

Comment (sabbyanandan) [reply content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md]: Cool with either of the choices, but I was only suggesting that we could follow 1 pattern consistently everywhere.

File: content/documentation/pages/1-installation/3-kubernetes/1-helm.md (hunk 6 above)
Comment (sabbyanandan) [reply content/documentation/pages/1-installation/3-kubernetes/1-helm.md]: I see. I was testing all my manifests with `value: |-`, so I wasn't sure whether what you have here would work. Did you try with `value: -|`, and it works?

**Comment by dturanski** [comment]
Rebased, squashed, and merged.

**Pull request closed without merging**